### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by jtaylorz**
Recombined the preliminary installation instructions for mac and linux-based distributions. Added a page for users running an analysis on a system that they do not have sudo privileges on that is essentially a list of necessary dependencies and packages. Should I explicitly state version numbers for packages like CMake? I am not sure what the minimum version would be and it seems better to just imply that newer versions should be used. Also, I have changed the the instructions to utilize the requirements.txt file for python package installs.

### CODE REVIEW

File: doc/prelims.rst [hunk 1]
```diff
@@ -22,39 +25,75 @@ Running a **Linux** distribution, Python must be compiled. Download and compile
...
+Running **macOS** install/update Python 3.0+ (here using the reccomended `Homebrew <https://brew.sh/>`__ package manager):
 
-2) Install virtualenvwrapper
-----------------------------
+.. code-block:: shell
+
+   brew install python3
+   brew upgrade python3 
+
+A .profile file is a shell script that Bash runs whenever it is started or executed by the user. Setting up pytpc on Linux distributions involves adding lines to a built-in .bashrc, and the .profile is the macOS equivalent. However, you must first create the .profile file as it is not included on Mac systems by default. **Create and open a plain text file titled `.profile.txt` in your home directory**. This file will not appear in your home directory unless you enter 'ls -a' in the terminal to show hidden files.
```
Comment (jbradt) [inline doc/prelims.rst]: It has to be just `.profile`, not `.profile.txt`.

File: doc/prelims.rst [hunk 2]
```diff
@@ -22,39 +25,75 @@ Running a **Linux** distribution, Python must be compiled. Download and compile
...
+
+.. code-block:: shell
+
+   brew install llvm
+
+To make this most recent version of clang the default compiler on your system (and enable OpenMP support) paste the following lines into the .profile file created above:
+
+.. code-block:: shell
+
+   export CC=/usr/local/Cellar/llvm/4.0.0_1/bin/clang-4.0
```
Comment (jbradt) [inline doc/prelims.rst]: Make these paths use the `$(brew --prefix llvm)` command from below. I might write it like this:
```shell
LLVM_PATH=$(brew --prefix llvm)
export CC=${LLVM_PATH}/bin/clang
export CXX=${LLVM_PATH}/bin/clang++
```
That way it doesn't depend on where it installed LLVM (which is `/usr/local/opt/llvm` on my system, actually), and it only runs the slow `brew --prefix llvm` command once.

File: doc/prelims.rst [hunk 3]
```diff
@@ -22,39 +25,75 @@ Running a **Linux** distribution, Python must be compiled. Download and compile
...
    
-   pip3.6 install virtualenvwrapper
+   pip install virtualenvwrapper   # sudo might be required
 
-An introduction and walkthrough to using the virtualenvwrapper tool is found `here <https://virtualenvwrapper.readthedocs.io/en/latest/>`__. To use virtual environments, place the following lines in the .bashrc file. The first line sets the Python interpreter for your virtual environments to python3.6. The last line is a path to your shell startup file, change it depending on where virtualenvwrapper was installed by pip.
+An introduction and walkthrough to using the virtualenvwrapper tool can be found `here <https://virtualenvwrapper.readthedocs.io/en/latest/>`__. To use virtual environments, place the following lines in your .bashrc or .profile file. The first line sets the Python interpreter for your virtual environments to python3.6. The last line is a path to your shell startup file and you should change it depending on where virtualenvwrapper was installed by pip.
 
 .. code-block:: shell
 
    VIRTUALENVWRAPPER_PYTHON=/usr/local/bin/python3.6
```
Comment (jbradt) [inline doc/prelims.rst]: This path is not correct on the Mac if you install Python from the dmg file from python.org. That installer puts it somewhere deep in `/Library/Frameworks`. Maybe we should write `VIRTUALENVWRAPPER_PYTHON=$(which python3.6)` instead.

**Comment by jtaylorz** [comment]
I have made these more flexible environment variable changes. I actually did copy the requirements.txt file, that's my mistake so I'm sorry about that. I'll be sure to merge in the future. Would you like me to remove it and merge for this pull request or just keep it in mind from now on?

**Comment by jbradt** [comment]
@jtaylorz No, I think we should just leave it as-is since it'll make the history really confusing if you delete the file and then merge it in again. So go ahead and push your changes, and I'll merge this.

**Pull request merged**